fem/fe_coll.cpp

```diff
@@ -289,6 +290,31 @@ FiniteElementCollection *FiniteElementCollection::New(const char *name)
    return fec;
 }
 
+FiniteElementCollection *FiniteElementCollection::Clone(int p) const
+{
+   // default implementation for collections that don't care about variable p
+   MFEM_ABORT("Collection " << Name() << " does not support variable orders.");
+   (void) p;
```

Comment:
What does `(void) p;` do?

Reply:
This is a common technique to suppress an "unused variable" compiler warning.

Reply:
Also it explicitly documents that we are not using the parameter and are aware of it.